lang/java/reef-checkpoint/src/main/java/org/apache/reef/io/checkpoint/RandomNameCNS.java

```diff
@@ -26,20 +26,31 @@
 import javax.inject.Inject;
 
 /**
- * Simple naming service that generates a random checkpoint name.
+ * A naming service that generates a random checkpoint name by appending a random alphanumeric string (suffix)
+ * of a given length to a user-supplied prefix string.
  */
 public class RandomNameCNS implements CheckpointNamingService {
 
   private final String prefix;
+  private final int lengthOfRandomSuffix;
+
 
   @Inject
   public RandomNameCNS(@Parameter(PREFIX.class) final String prefix) {
     this.prefix = prefix;
+    this.lengthOfRandomSuffix = 8;
```

Comment:
>>we can probably drop the `@Inject` on this constructor
> Doesn't this break the public API w.r.t. Tang ? 

Not if we have a default value, no.

> Is it OK to do the following ? 
> `this.lengthOfRandomSuffix = Integer.parseInt(LengthOfRandomSuffix.class.getAnnotation(NamedParameter.class).default_value());`

Yes. Alternatively, you can define a `public static final DEFAULT_VALUE` in `LengthOfRandomSuffix`. I've done that in the past.